### PR TITLE
Site id term refactor

### PR DESCRIPTION
**Pull request by dzucker-tab**
Rationalizing and clarifying uses of "site-id" and related terms.

### CODE REVIEW

File: docs/api-ref.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ Name | Description
 :--- | :---   
 `username` | The name of the user whose credentials will be used to sign in.   
 `password` | The password of the user.     
-`site_id` | This corresponds to the `contentUrl` attribute in the Tableau REST API. The `site_id` is the portion of the URL that follows the `/site/` in the URL. For example, "MarketingTeam" is the `site_id` in the following URL *MyServer*/#/site/**MarketingTeam**/projects. To specify the default site on Tableau Server, you can use an empty string `''`  (single quotes, no space).  For Tableau Online, you must provide a value for the `site_id`.  
+`site_id` | This corresponds to the `contentUrl` attribute in the Tableau REST API. It is the portion of the URL that follows the `/site/` in the URL. For example, "MarketingTeam" is the `site_id` in the following URL: `https://MyServer/#/site/MarketingTeam/projects`. To specify the default site on Tableau Server, you can either omit the site_id or use an empty string `''`  (single quotes, no space).  For Tableau Online, you must provide a value for the `site_id`.
```
Comment (jacalata) [inline docs/api-ref.md:53]: can we change this from 'site_id' to site_url_id or something? Then all the other uses of id should be luids.

File: docs/api-ref.md [hunk 2]
```diff
@@ -61,7 +61,7 @@ Source file: models/tableau_auth.py
 import tableauserverclient as TSC
 # create a new instance of a TableauAuth object for authentication
 
-tableau_auth = TSC.TableauAuth('USERNAME', 'PASSWORD', site_id='CONTENTURL')
+tableau_auth = TSC.TableauAuth('username', 'password', 'site-id')
```
[the remark this answers is not in the document]
Comment (jacalata) [reply docs/api-ref.md:64]: can you add back the keywords here?

File: docs/api-ref.md [hunk 3]
```diff
@@ -2249,7 +2249,7 @@ Returns a list of all `SiteItem` objects and a `PaginationItem`. Use these value
 #### sites.get_by_id
 
 ```py
-sites.get_by_id(site_id)
+sites.get_by_id(site_luid)
```
[the remark this answers is not in the document]
Comment (jacalata) [reply docs/api-ref.md:2252]: Agreed, should just be site_id again

[the remark this answers is not in the document]
Comment (jacalata) [reply docs/api-ref.md:2252]: Also on line 2264 below, 2402, 2415, 2424,
perhaps we could add an example value here, e.g 3d512018-64c4-4b66-a1fa-1c83710e323c

File: docs/api-ref.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jacalata) [reply docs/api-ref.md:64]: also line 3497, 3609

File: docs/index.md [hunk 4]
```diff
@@ -76,14 +76,14 @@ Run the following code to get a list of all the data sources on your installatio
 ```py
 import tableauserverclient as TSC
 
-tableau_auth = TSC.TableauAuth('USERNAME', 'PASSWORD', 'SITENAME')
+tableau_auth = TSC.TableauAuth('username', 'password', 'site-id')
```
[the remark this answers is not in the document]
Comment (jacalata) [reply docs/index.md:79]: and this

**Review by jacalata — CHANGES_REQUESTED** [review]

**Comment by bcantoni** [comment]
This one hasn't been touched in years - closing.

**Pull request closed without merging**